engines/wintermute/ad/ad_scene_geometry.cpp

```diff
@@ -960,121 +960,103 @@ bool AdSceneGeometry::createLights() {
+		if (_lights[i]->_active) {
+			++activeLightCount;
+		}
+	}
 
-	//	int NumActiveLights = 0;
-	//	for(int i=0; i<_lights.size(); i++)
-	//	{
-	//		_lights[i]->m_IsAvailable = false;
-	//		if(_lights[i]->m_Active) NumActiveLights++;
-	//	}
-	//	if(NumActiveLights <= MaxLights)
-	//	{
-	//		for(int i=0; i<_lights.size(); i++)
-	//		{
-	//			_lights[i]->m_IsAvailable = true;
-	//		}
-	//	}
-	//	else
-	//	{
-	//		if(!m_MaxLightsWarning)
-	//		{
-	//			_gameRef->LOG(0, "Warning: Using more lights than the hardware supports (%d)", MaxLights);
-	//			m_MaxLightsWarning = true;
-	//		}
+	if (activeLightCount <= _gameRef->_renderer3D->maximumLightsCount()) {
+		for (uint i = 0; i < _lights.size(); i++) {
+			_lights[i]->_isAvailable = true;
+		}
+	} else {
+		if (!_maxLightsWarning) {
+			_gameRef->LOG(0, "Warning: Using more lights than the hardware supports (%d)", _gameRef->_renderer3D->maximumLightsCount());
+			_maxLightsWarning = true;
+		}
 
-	//		BaseArray<Light3D*> ActiveLights;
+		Common::Array<Light3D *> activeLights;
 
-	//		// compute distance to point
-	//		for(int i=0; i<_lights.size(); i++)
-	//		{
-	//			if(!_lights[i]->m_Active) continue;
+		// compute distance to point
+		for (uint i = 0; i < _lights.size(); i++) {
+			if (!_lights[i]->_active) {
+				continue;
+			}
 
-	//			Math::Vector3d Dif;
-	//			if(_lights[i]->m_IsSpotlight)
-	//			{
-	//				//Dif = _lights[i]->m_Target - Point;
-	//				Math::Vector3d Dir = _lights[i]->m_Target - _lights[i]->m_Pos;
-	//				Dif = (_lights[i]->m_Pos + Dir * 0.75f) - Point;
-	//			}
-	//			else
-	//				Dif = _lights[i]->m_Pos - Point;
+			Math::Vector3d dif;
 
-	//			_lights[i]->m_Distance = fabs(D3DXVec3Length(&Dif));
+			if (_lights[i]->_isSpotlight) {
+				Math::Vector3d dir = _lights[i]->_target - _lights[i]->_position;
+				dif = (_lights[i]->_position + dir * 0.75f) - point;
+			} else {
+				dif = _lights[i]->_position - point;
+			}
 
-	//			ActiveLights.add(_lights[i]);
-	//		}
+			_lights[i]->_distance = dif.getMagnitude();
 
-	//		// sort by distance
-	//		if(ActiveLights.size() > 0)
-	//		{
-	//			qsort(ActiveLights.GetData(), ActiveLights.size(), sizeof(Light3D*), AdSceneGeometry::CompareLights);
+			activeLights.push_back(_lights[i]);
+		}
 
-	//			for(int i=0; i<ActiveLights.size(); i++)
-	//			{
-	//				ActiveLights[i]->m_IsAvailable = i < MaxLights;
-	//			}
-	//		}
-	//	}
+		// sort by distance
+		if (activeLights.size() > 0) {
+			Common::sort(activeLights.begin(), activeLights.end(), compareLights);
 
-	//	// light all available lights
-	//	for(int i=0; i<100; i++)
-	//	{
-	//		m_Renderer->m_Device->LightEnable(i, FALSE);
-	//	}
+			for (uint i = 0; i < activeLights.size(); i++) {
+				activeLights[i]->_isAvailable = static_cast<int>(i) < _gameRef->_renderer3D->maximumLightsCount();
+			}
+		}
+	}
 
-	//	NumActiveLights = 0;
-	//	for(int i=0; i<_lights.size(); i++)
-	//	{
-	//		if(NumActiveLights >= MaxLights) break;
+	// light all available lights
```

Comment:
Slightly weird comment, given that the code immediately below it disables lights.